crownstone_uart/core/CrownstoneUart.py

```diff
@@ -71,18 +71,16 @@ def handleMessage(result, data):
         while not result[0] and self.running:
             try:
                 exc = self.manager_exception_queue.get(block=False)
-                # log error & wait for thread to close
-                _LOGGER.warning(f"Error occurred while initializing USB: {exc}, quitting.")
                 self.uartManager.join()
                 self.stop()
-                break
+                raise exc[0](exc[1])
```

Comment:
This looks dodgy, is there no way to improve on this? Maybe add something other than `sys.exc_info()` in the queue.
Maybe also rename `exc` to `exception`.

Reply:
`sys.exc_info()` holds all the info the the exception raised in the thread, this includes:

* type (UartException in this case)
* value (UartBridgeError.CANNOT_OPEN_SERIAL_CONTROLLER)
* traceback (not used)

I currently wouldn't know what else to use here, but suggestions are welcome!

Reply:
I guess you could read it out as tuple, something like: `exception, exception_value, trace = bridge_exception_queue.get(block=False)`